traffic_ops/app/bin/checks/ToDSCPCheck.pl

```diff
@@ -105,179 +111,231 @@
 
 my $check_name = $jconf->{check_name};
 if ( $check_name ne "DSCP" ) {
-   ERROR "This Check Extension is exclusively for DSCP.";
+   FATAL "FATAL This Check Extension is exclusively for DSCP.";
    print "\n\n";
    &help();
    exit(4);
 }
 
-
-TRACE( "force: " . $args{f} . "" );
-
-TRACE Dumper($jconf);
+DEBUG "DEBUG " . Dumper($jconf);
 my $b_url = $jconf->{base_url};
 Extensions::Helper->import();
 my $ext = Extensions::Helper->new( { base_url => $b_url, token => '91504CE6-8E4A-46B2-9F9F-FE7C15228498' } );
 
 my %ds_info           = ();
 my $jdeliveryservices = $ext->get( Extensions::Helper::DSLIST_PATH );
 
-# Get all the Deliver Services
+# Get all the Delivery Services
 foreach my $ds ( @{$jdeliveryservices} ) {
    $ds_info{ $ds->{id} } = $ds;
+
+   # Get the DS details and add the matchList
+   my $ds_details = $ext->get( '/api/1.2/deliveryservices/' . $ds->{id} );
+   #DEBUG "Debug matchList before push: " . Dumper($ds_details->[0]{matchList});
+   # The matchList was removed at one point and now it's back.
+   # Leaving this here for when that happens again.
+   #push( @{ $ds_info{ $ds->{id} }{matchList} }, @{ $ds_details->[0]{matchList} } );
+   #DEBUG "Debug matchList after push: " . Dumper($ds_info{ $ds->{id} }{matchList});
+}
+
+my $jdataserver = ();
+if (defined($args{s})) {
+   # TODO check the return on this
+   DEBUG "DEBUG Getting single server";
+   # This is a reference to a hash
+   $jdataserver = $ext->get( '/api/1.1/servers/hostname/' . $args{s} . '/details' );
+
+   # create an array
+   my @tmp = ();
+   # put the hash reference in the array
+   $tmp[0] = $jdataserver;
+   # clear $jdataserver
+   $jdataserver = ();
+   # create a reference to the array
+   $jdataserver = \@tmp;
+
+} else {
+   # This is a reference to an array
+   $jdataserver = $ext->get( Extensions::Helper::SERVERLIST_PATH );
+   unless (ref($jdataserver)) {
+      print "jdataserver is not a reference at all.\n";
+   }
+   if (ref($jdataserver) eq "SCALAR") {
+      print "jdataserver is a reference to a scalar.\n";
+   }
+   if (ref($jdataserver) eq "ARRAY") {
+      print "jdataserver is a reference to a array.\n";
+   }
+   if (ref($jdataserver) eq "HASH") {
+      print "jdataserver is a reference to a hash.\n";
+   }
+}
+# exit ();
```

Comment:
Looks like you left some debugging code in here - if you really want to print that stuff out, shouldn't it be with `DEBUG`? (Also that commented `exit ();` doesn't really need to be there)